File: 백준/Silver/10828. 스택/스택.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let N;
const stack = [];
let count = 0;
const output = [];

rl.on('line', (line) => {
    if (!N) {
        N = parseInt(line);
        return;
    }

    const [instruction, value] = line.split(' ');

    switch (instruction) {
        case 'push':
            stack.push(value);
            break;
        case 'pop':
            output.push(stack.length ? stack.pop() : -1);
            break;
        case 'top':
            output.push(stack.length ? stack[stack.length - 1] : -1);
            break;
        case 'size':
            output.push(stack.length);
            break;
        case 'empty':
            output.push(stack.length === 0 ? 1 : 0);
            break;
    }

    count++;
    if (count === N) {
        console.log(output.join('\n'));
        rl.close();
    }
});

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

N = int(input())

commands = [input().split() for _ in range(N)]
stack = []

for command in commands:
  if command[0] == 'push':
    value = command[1]
    stack.append(value)
  elif command[0] == 'pop':
    if len(stack) == 0:
      print(-1)
    else:
      print(stack.pop())
  elif command[0] == 'size':
    print(len(stack))
  elif command[0] == 'empty':
    print(1) if len(stack) == 0 else print(0)
  elif command[0] == 'top':
    if len(stack) == 0:
      print(-1)
    else:
      print(stack[-1])